apps/system/flash_test/flash_test.c

```diff
@@ -84,18 +351,55 @@ static int flash_read(unsigned long addr, int size)
 	int remaining;
 	FAR uint8_t *buffer;
 	FAR struct mtd_dev_s *dev_mtd = NULL;
+	long long total_duration_ms = 0;
+	systick_time_ms = 0;
 
 	if (addr == 0 || size <= 0) {
 		printf("Invalid arguments, offset 0x%x, read size %d\n", addr, size);
 		return ERROR;
 	}
+	if (flash == FLASH_INTERNAL) {
+		printf("Received Flash Read cmd for internal flash offset 0x%x, read size %d\n", addr, size);
+		dev_mtd = up_flashinitialize();
```

Comment:
Please check below comment

Reply:
using the flash driver for the flat app config. update the app accordingly